### PR TITLE
Fix Ingress controller broken link

### PR DESCRIPTION
**Pull request by mfpierre**

### CODE REVIEW

**Comment by mfpierre** [comment]
I just signed the CLA

File: contributors/design-proposals/architecture/architecture.md [hunk 1]
```diff
@@ -237,7 +237,7 @@ Service endpoints are found primarily via [DNS](https://kubernetes.io/docs/conce
 A number of components, called [*add-ons*](https://github.com/kubernetes/kubernetes/tree/master/cluster/addons) typically run on Kubernetes
 itself:
 * [DNS](https://github.com/kubernetes/kubernetes/tree/master/cluster/addons/dns)
-* [Ingress controller](https://github.com/kubernetes/ingress/tree/master/controllers)
+* [Ingress controller](https://github.com/kubernetes/ingress/)
```
Comment (xiangpengzhao) [inline contributors/design-proposals/architecture/architecture.md]: Seems like it should be `https://github.com/kubernetes/ingress-nginx/tree/master/pkg/ingress/controller`?

Comment (mfpierre) [reply contributors/design-proposals/architecture/architecture.md]: Looking at old link you're probably right, I changed it

**Comment by xiangpengzhao** [comment]
@mfpierre also note that the original ingress repo has been split to two repos (ingress-nginx and ingress-gce), I'm not sure if we should add some description where you touched in this PR.
/cc @freehan

**Pull request merged**